protocols/kad/src/protocol.rs

```diff
@@ -171,16 +167,30 @@ where
     }
 }
 
-impl<C> OutboundUpgrade<C> for KademliaProtocolConfig
+/// Configuration for a Kademlia outbound connection upgrade. When applied to a connection, turns this
+/// connection into a `Stream + Sink` whose items are of type `KadRequestMsg` and `KadResponseMsg`.
+// TODO: if, as suspected, we can confirm with Protocol Labs that each open Kademlia substream does
+//       only one request, then we can change the output of the `Upgrade` to be just a single message
```

Comment:
Do we have an issue open with them for this? It'd be good to add a reference to it here.